### PR TITLE
[RA] Have removed the toLowerCase that was forcing all URLs to lowerc…

### PR DESCRIPTION
**Pull request by bearsoftware**
Hi,
I have implemented the feature request https://community.smartbear.com/t5/SoapUI-Feature-Requests/Allow-uppercase-in-URL-in-the-HTTP-Request-test-step/idi-p/105272 by making a very simple change and associated unit test.
Since this is my first attempt at a pull request against this project, please let me know if there is anything else that you would like me to do / document? 
Thanks for your help,
Rupert Anderson

### CODE REVIEW

**Comment by olensmar** [comment]
Hi!

Awesome - thank you so much! We need to ask you to send the signed CLA as described at https://www.soapui.org/developers-corner/contribute-to-soapui.html before we can merge this - but apart from that you're all set!

/Ole

File: soapui/src/main/java/com/eviware/soapui/impl/support/HttpUtils.java [hunk 1]
```diff
@@ -53,7 +53,7 @@ public static String completeUrlWithHttpIfProtocolIsNotHttpOrHttpsOrPropertyExpa
         if (StringUtils.isNullOrEmpty(endpoint)) {
             return endpoint;
         }
-        endpoint = endpoint.toLowerCase().trim();
+        endpoint = endpoint.trim();
```
Comment (olensmar) [inline soapui/src/main/java/com/eviware/soapui/impl/support/HttpUtils.java]: I'm guessing this will fail for endpoints that are (mistakenly) entered with capitals - i.e. "HTTP://" - which would result in the returned endpoint being "http://HTTP://..." - perhaps you can handle this and add a corresponding unit-test!? Or perhaps I'm missing something!?

Comment (bearsoftware) [inline soapui/src/main/java/com/eviware/soapui/impl/support/HttpUtils.java]: Hi Ole,
Its an honour to hear from you! :-)
I sent my signed CLA to Manne - shall I send it to him again? I don't have your email address.
You are of course right about the possibility of someone entering HTTP - I'll add a test & fix for that - thanks for spotting that, good catch!
Thanks for your help,
Rupert

Comment (bearsoftware) [inline soapui/src/main/java/com/eviware/soapui/impl/support/HttpUtils.java]: Hi Ole,
I made the changes and tests to cover the URL beginning with HTTP/HTTPS i.e.
`  
...
endpoint = endpoint.trim();
String lowerCaseEndpoint = endpoint.toLowerCase();
if (!lowerCaseEndpoint.startsWith("http://") && !lowerCaseEndpoint.startsWith("https://") && !endpoint.startsWith("$")) {
            return "http://" + endpoint;
}
...`
However, maybe as a second thought and counterpoint from a usability perspective, I would say that whilst this part of the code tries to help by automatically appending http://, it is actually potentially irritating in cases where the user is manually editing the protocol in the TestStep e.g. a bit like an over eager spellchecker that shoves in an unwanted correction! 
Basically that HTTP Request TestStep is a gem mainly because it is (almost) free format and lacks the structural restrictions on the REST Request TestStep, most people just want full control over the endpoint URL. So, what do you think about me instead refactoring the code to just trim the URL i.e. just

`if (StringUtils.isNullOrEmpty(endpoint)) {
            return endpoint;
  }
  endpoint = endpoint.trim();`

Let me know your thoughts.

Thanks for your time,
Rupert

**Comment by bearsoftware** [comment]
Hi Ole,
As we are both busy guys, rather than discuss any alternatives, I have pushed the fix & test that you originally spotted.
Please let me know if this is OK? Or if you need anything else?
I am very keen to get this fix / improvement in before I invest anymore time.
Thanks for your help,
Rupert

**Comment by olensmar** [comment]
Hi @bearsoftware - I'm sorry this has been lingering - looks good to me now!

**Comment by bearsoftware** [comment]
Thanks, thats OK, I know you must be very busy :-)
I hope to contribute more assuming this change works out and gets merged in, so if theres anything I can do to 'oil the cogs' then please let me know. I might even blog about it to hopefully help encourage others to contribute!
Thanks for your help,
Rupert
